crates/fuel-core/src/executor.rs

```diff
@@ -3038,6 +3040,485 @@ mod tests {
+                .collect::<Vec<_>>();
+
+            relayer_db_for_events(&events, da_height)
+        }
+
+        #[test]
+        fn execute_without_commit__invalid_relayed_txs_are_not_included_and_are_reported()
+        {
+            let genesis_da_height = 3u64;
+            let block_height = 1u32;
+            let da_height = 10u64;
+
+            // given
+            let relayer_db = relayer_db_with_invalid_relayed_txs(da_height);
+
+            // when
+            let on_chain_db = database_with_genesis_block(genesis_da_height);
+            let producer = create_relayer_executor(on_chain_db, relayer_db);
+            let block = test_block(block_height.into(), da_height.into(), 0);
+            let (result, _) = producer
+                .execute_without_commit(ExecutionTypes::Production(block.into()))
+                .unwrap()
+                .into();
+
+            // then
+            let txs = result.block.transactions();
+            assert_eq!(txs.len(), 1);
+
+            // and
+            let events = result.events;
+            let fuel_core_types::services::executor::Event::ForcedTransactionFailed {
+                failure: actual,
+                ..
+            } = &events[0]
+            else {
+                panic!("Expected `ForcedTransactionFailed` event")
+            };
+            let expected = "Failed validity checks";
+            assert_eq!(expected, actual);
+        }
+
+        fn relayer_db_with_invalid_relayed_txs(da_height: u64) -> Database<Relayer> {
+            let event = arb_invalid_relayed_tx_event();
+            relayer_db_for_events(&[event], da_height)
+        }
+
+        #[test]
+        fn execute_without_commit__relayed_tx_with_low_max_gas_fails() {
+            let genesis_da_height = 3u64;
+            let block_height = 1u32;
+            let da_height = 10u64;
+
+            // given
+            let relayer_db = relayer_db_with_relayed_tx_with_low_max_gas(da_height);
+
+            // when
+            let on_chain_db = database_with_genesis_block(genesis_da_height);
+            let producer = create_relayer_executor(on_chain_db, relayer_db);
+            let block = test_block(block_height.into(), da_height.into(), 0);
+            let (result, _) = producer
+                .execute_without_commit(ExecutionTypes::Production(block.into()))
+                .unwrap()
+                .into();
+
+            // then
+            let txs = result.block.transactions();
+            assert_eq!(txs.len(), 1);
+
+            // and
+            let events = result.events;
+            let fuel_core_types::services::executor::Event::ForcedTransactionFailed {
+                failure: actual,
+                ..
+            } = &events[0]
+            else {
+                panic!("Expected `ForcedTransactionFailed` event")
+            };
+            let expected = "Insufficient max gas.";
+            assert!(actual.starts_with(expected));
```

Comment:
Instead of using hardcoded strings you can generate them from corresponding error variant.

```rust
assert_eq!(ForcedTransactionFailure::MaxGas.to_string(), actual);
```

The same applies to all tests

Reply:
I should have been clearer:
```
            assert!(actual.starts_with(expected));
 ```
 Is not an `assert_eq` it's checking that the start of the message is right.
 
 I can rename the `expected` to something better.